packages/next/next-server/lib/router/router.ts

```diff
@@ -818,12 +818,13 @@ export default class Router implements BaseRouter {
       if (process.env.NODE_ENV !== 'production') {
         return
       }
+
       const route = removePathTrailingSlash(pathname)
-      Promise.all([
+      await Promise.all([
         this.pageLoader.prefetchData(url, asPath),
         this.pageLoader[options.priority ? 'loadPage' : 'prefetch'](route),
-      ]).then(() => resolve(), reject)
-    })
+      ])
+    } catch (err) {}
```

Comment:
I don't think the `catch` is necessary, `.then(() => resolve(), reject)` would have made this function reject.